Widget concept documentation

ikas opened the pull request:
## Summary

This PR refers to [GTC-278](https://gfw.atlassian.net/browse/GTC-278). It adds a new section to the concept docs regarding **Widgets**.

## Testing instructions

If you feel like deep-diving into the contents of the PR, follow the steps below:

1. Clone the documentation repository and follow the [installation instructions on the README](https://github.com/resource-watch/doc-api#installation-and-usage).
2. Checkout the branch of this PR: `docs/widget-concept-docs`
3. Read the added concept docs for **widgets**.

ethanroday commented on source/includes/concept/_widget.md:
```diff
@@ -0,0 +1,17 @@
+## Widget
+
+You have fetched your data from your [dataset](#dataset) using [queries](#query). And you know how to build custom visualizations for geospatial data using [layers](#layer). Yet, sometimes your data is not geo-located, and you want to build a plain graphical representation of it: this is when **widgets** come in handy.
```
You can still use widgets for geo-referenced data - should clarify that it can be used for any kind of data, georeferenced or not.

ikas replied:
Clarified, thanks for pointing out :+1:

ethanroday commented on source/includes/concept/_widget.md:
```diff
@@ -0,0 +1,17 @@
+
+As with layers, each widget has a single dataset associated with it, and a dataset can be associated with many widgets. You can represent the same data in different ways by creating different widgets for the same dataset. The same widget can store independent configuration values for each RW API based application. It can also contain the required configuration for rendering the same visualization ...
+
+Like layers, the widget itself does not interact with the dataset data. You can either use the widget's `queryUrl` field to store the query to get the widget's data or store it inside the free form `widgetConfig` object. In any of these cases, it is your responsibility as an API user to query the data that will be used for rendering the widget.
```
If the widget doesn't interact with dataset data, why do I have to associate widgets with datasets? If the answer is "for organizational purposes" we should say that.

ikas replied:
Added a line at the start of the paragraph stating that the association between datasets and widgets is only for organizational purposes.

ethanroday commented on source/includes/concept/_widget.md:
```diff
@@ -0,0 +1,17 @@
+
+As in the case of layers (where many fields are free-form), the RW API does not apply any restriction to whatever is saved in the widget configuration field (`widgetConfig`). This allows for a very high level of flexibility for users, but has the downside of making it harder to document how to use widgets.
+
+However, contrary to what happens with layers, many of the existing widgets base themselves in the [Vega grammar](https://github.com/vega/vega) for configuring its graphical representations. Vega is a visualization grammar, a declarative format for creating interactive visualization designs. You can then use a chart library that supports the Vega grammar (such as [d3.js](https://d3js.org/)) to re ...
```
I would take a stronger stance here and say that Vega is the standard convention, but that you can use anything you want.
Also, really like that you describe the full use case here, even if it's just a couple of sentences. Exactly what concept docs are for!

ikas replied:
Reworded things a bit to say that the existing widgets' standard approach for building widget configuration objects is using Vega. Also, added a note at the end saying that users can still use whatever configuration format they wish :+1:

ikas commented:
> See individual comments for some requested changes. Otherwise looks good!

@ethanroday thank you for your feedback :bow: I have addressed all your comments, the PR should be ready to be merged :+1:
